src/store/dashboardStore.ts

```diff
@@ -0,0 +1,30 @@
+import { create } from 'zustand';
+import { persist } from 'zustand/middleware';
+import type { Dashboard } from '@/types/dashboards';
+import { getBoard } from '@/lib/boardService';
+
+interface dashboardState {
+  dashboard: Dashboard | null;
+  setDashboard: (dashboardId: number) => void;
+}
+
+const useDashboardStore = create(
+  persist<dashboardState>(
+    (set, get) => ({
+      dashboard: null,
+      setDashboard: async (dashboardId) => {
+        if (get().dashboard?.id === dashboardId) return;
+
+        const response = await getBoard(dashboardId.toString());
+        set({
+          dashboard: { ...response },
+        });
+      },
+    }),
+    {
+      name: 'dashboardStorage',
+    }
```

Comment:
```suggestion
    {
      name: 'dashboardStorage',
      partialize: (state) => ({
      dashboard: state.dashboard,
    }),
  }
```
이렇게 작성하면 로컬 스토리지의 데이터가 변경이 있을 때만 업데이트된다네요?!

Reply:
partialize 요거 특정 필드만 저장하고 싶을데 쓰는게 아닌가요..?! 아래 내용처럼 store에는 2개 필드가 있는데 1개필드만 로컬스토리지에 저장하고싶을때..!
```
export const useBoundStore = create(
  persist(
    (set, get) => ({
      foo: 0,
      bar: 1,
    }), {
      partialize: (state) => ({
        foo: state.foo,
      }),
    }
  )
);
```